presidio-image-redactor/presidio_image_redactor/dicom_image_redactor_engine.py

```diff
@@ -733,12 +737,74 @@ def _add_redact_box(
 
         return redacted_instance
 
+    def _get_analyzer_results(
+        self,
+        image: PIL.PngImagePlugin.PngImageFile,
+        instance: pydicom.dataset.FileDataset,
+        redact_approach: Union[str, PatternRecognizer],
+        ocr_kwargs: Optional[dict],
+        **text_analyzer_kwargs
+    ) -> List[ImageRecognizerResult]:
+        """Analyze image with selected redaction approach.
+
+        :param image: DICOM pixel data as PIL image.
+        :param instance: DICOM instance (with metadata).
+        :param redact_approach: What approach to use when redacting
+        ("default", "metadata", "allow", or a PatternRecognizer object).
```

Comment:
Please see my comment on this approach further down. In case we decide to go with this approach, could you please provide more information on each redact approach in the docstring? (Here and in the main function's docstring).